### PR TITLE
netrw: move netrw_home to XDA_DATA_HOME

### LINKED ISSUE #12161: move "g:netrw_home" to $XDA_DATA_HOME/nvim/
- `nvim --version`: 0.4.3
- `vim -u DEFAULTS` (version: 8.1) behaves differently? yes - it puts it in ~/.vim/
- Operating system/version: macOS 10.15.3
- Terminal name/version: iTerm2 3.3.9
- `$TERM`: screen-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC

:Explore
< move somewhere across directories >
< close neovim >
< go to $XDA_CONFIG_HOME/nvim/ >
< ls -a >
< .netrwhist file is there >
```

### Actual behaviour

`.netrwhist` by default is created in `$XDA_CONFIG_HOME/nvim/`. To me it looks like it doesn't belong there. It is not config data, but rather a dump representing a variable state.

### Expected behaviour

Change default setting of `g:netrw_home` to `$XDA_DATA_HOME/nvim/ ` so that`.netrwhist` is placed among other types of variable data.

### PR DESCRIPTION
**Pull request by doubleloop**
fixes #12161

### CODE REVIEW

File: runtime/doc/pi_netrw.txt [hunk 1]
```diff
@@ -2745,8 +2745,7 @@ your browsing preferences.  (see also: |netrw-settings|)
 				history are saved (as .netrwbook and
 				.netrwhist).
 				Netrw uses |expand()|on the string.
-				 default: the first directory on the
-				         |'runtimepath'|
+				 default: ~/.local/share/nvim/ (|xdg|)
```
Comment (janlazo) [inline runtime/doc/pi_netrw.txt:2748]: This is the wrong directory if not using xdg. Stick to `stdpath('data')`.

**Comment by doubleloop** [comment]
I believe test fail is not related to this change:
`[  FAILED  ] test/functional\ui\mode_spec.lua @ 48: ui mode_change event works in insert mode`

**Review by janlazo — APPROVED** [review]
I prefer the netrw are stored within their own directory like shada files but this is how netrw works :disappointed: 

Good job.

**Comment by janlazo** [comment]
@doubleloop Can you check the unmerged Vim runtime files if porting netrw runtimes causes any conflicts to your PR? I prefer to merge to this after updating netrw as a whole.

**Comment by janlazo** [comment]
Will merge after https://github.com/neovim/neovim/pull/13949

**Comment by janlazo** [comment]
Thanks.

**Pull request merged**